### ISSUE TITLE
ces deployment's name can not be set to other than "ces-controller"

### ISSUE THREAD

**Issue by yxq-yxq** [f5devcentral/container-egress-service#5]
when applying ces deployment, if the ces deployment's name is not "ces-controller", the ces pod will go to "Error" state with log "failed to get deploy[ces-controller]".  Here are the details： 

1. ces image：f5devcentral/ces-controller:0.5.2
2. error log：
```
W1229 09:48:44.552385       1 client_config.go:614] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
F1229 09:48:44.606279       1 main.go:108] failed to get deploy[ces-controller]
goroutine 1 [running]:
k8s.io/klog/v2.stacks(0xc0001a8001, 0xc0000c4000, 0x50, 0xe2)
        /home/runner/work/container-egress-service/container-egress-service/vendor/k8s.io/klog/v2/klog.go:1026 +0xb9
k8s.io/klog/v2.(*loggingT).output(0x222a280, 0xc000000003, 0x0, 0x0, 0xc0000ba000, 0x0, 0x1bc7aaa, 0x7, 0x6c, 0x0)
        /home/runner/work/container-egress-service/container-egress-service/vendor/k8s.io/klog/v2/klog.go:975 +0x1e5
k8s.io/klog/v2.(*loggingT).printf(0x222a280, 0xc000000003, 0x0, 0x0, 0x0, 0x0, 0x171685e, 0x18, 0xc0000ff1f0, 0x1, ...)
        /home/runner/work/container-egress-service/container-egress-service/vendor/k8s.io/klog/v2/klog.go:753 +0x19a
k8s.io/klog/v2.Fatalf(...)
        /home/runner/work/container-egress-service/container-egress-service/vendor/k8s.io/klog/v2/klog.go:1514
main.main()
        /home/runner/work/container-egress-service/container-egress-service/cmd/ces/main.go:108 +0x1379

goroutine 18 [chan receive]:
k8s.io/klog/v2.(*loggingT).flushDaemon(0x222a280)
        /home/runner/work/container-egress-service/container-egress-service/vendor/k8s.io/klog/v2/klog.go:1169 +0x8b
created by k8s.io/klog/v2.init.0
        /home/runner/work/container-egress-service/container-egress-service/vendor/k8s.io/klog/v2/klog.go:420 +0xdf

goroutine 33 [syscall]:
os/signal.signal_recv(0x0)
        /opt/hostedtoolcache/go/1.16.12/x64/src/runtime/sigqueue.go:168 +0xa5
os/signal.loop()
        /opt/hostedtoolcache/go/1.16.12/x64/src/os/signal/signal_unix.go:23 +0x25
created by os/signal.Notify.func1.1
        /opt/hostedtoolcache/go/1.16.12/x64/src/os/signal/signal.go:151 +0x45

goroutine 34 [chan receive]:
github.com/kubeovn/ces-controller/pkg/signals.SetupSignalHandler.func1(0xc0003bdb60, 0xc000182d80)
        /home/runner/work/container-egress-service/container-egress-service/pkg/signals/signal.go:36 +0x34
created by github.com/kubeovn/ces-controller/pkg/signals.SetupSignalHandler
        /home/runner/work/container-egress-service/container-egress-service/pkg/signals/signal.go:35 +0xd1

goroutine 7 [IO wait]:
internal/poll.runtime_pollWait(0x7fcbab29c498, 0x72, 0xffffffffffffffff)
        /opt/hostedtoolcache/go/1.16.12/x64/src/runtime/netpoll.go:227 +0x55
internal/poll.(*pollDesc).wait(0xc0000b4218, 0x72, 0x900, 0x9ca, 0xffffffffffffffff)
        /opt/hostedtoolcache/go/1.16.12/x64/src/internal/poll/fd_poll_runtime.go:87 +0x45
internal/poll.(*pollDesc).waitRead(...)
        /opt/hostedtoolcache/go/1.16.12/x64/src/internal/poll/fd_poll_runtime.go:92
internal/poll.(*FD).Read(0xc0000b4200, 0xc00012c000, 0x9ca, 0x9ca, 0x0, 0x0, 0x0)
        /opt/hostedtoolcache/go/1.16.12/x64/src/internal/poll/fd_unix.go:166 +0x1d5
net.(*netFD).Read(0xc0000b4200, 0xc00012c000, 0x9ca, 0x9ca, 0x906, 0xc00012c0bf, 0x5)
        /opt/hostedtoolcache/go/1.16.12/x64/src/net/fd_posix.go:55 +0x4f
net.(*conn).Read(0xc00000e010, 0xc00012c000, 0x9ca, 0x9ca, 0x0, 0x0, 0x0)
        /opt/hostedtoolcache/go/1.16.12/x64/src/net/net.go:183 +0x91
crypto/tls.(*atLeastReader).Read(0xc00000c600, 0xc00012c000, 0x9ca, 0x9ca, 0x906, 0xc000088000, 0x0)
        /opt/hostedtoolcache/go/1.16.12/x64/src/crypto/tls/conn.go:776 +0x63
bytes.(*Buffer).ReadFrom(0xc000112278, 0x18cd680, 0xc00000c600, 0x40b6a5, 0x156f560, 0x16a9340)
        /opt/hostedtoolcache/go/1.16.12/x64/src/bytes/buffer.go:204 +0xbe
crypto/tls.(*Conn).readFromUntil(0xc000112000, 0x18cf920, 0xc00000e010, 0x5, 0xc00000e010, 0x96)
        /opt/hostedtoolcache/go/1.16.12/x64/src/crypto/tls/conn.go:798 +0xf3
crypto/tls.(*Conn).readRecordOrCCS(0xc000112000, 0x0, 0x0, 0x0)
        /opt/hostedtoolcache/go/1.16.12/x64/src/crypto/tls/conn.go:605 +0x115
crypto/tls.(*Conn).readRecord(...)
        /opt/hostedtoolcache/go/1.16.12/x64/src/crypto/tls/conn.go:573
crypto/tls.(*Conn).Read(0xc000112000, 0xc00014f000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        /opt/hostedtoolcache/go/1.16.12/x64/src/crypto/tls/conn.go:1276 +0x165
bufio.(*Reader).Read(0xc00014a120, 0xc0001382d8, 0x9, 0x9, 0x8eea6b, 0xc0004a9c78, 0x406fa5)
        /opt/hostedtoolcache/go/1.16.12/x64/src/bufio/bufio.go:227 +0x222
io.ReadAtLeast(0x18cd520, 0xc00014a120, 0xc0001382d8, 0x9, 0x9, 0x9, 0xc000175100, 0x14c3f45ad5db00, 0xc000175100)
        /opt/hostedtoolcache/go/1.16.12/x64/src/io/io.go:328 +0x87
io.ReadFull(...)
        /opt/hostedtoolcache/go/1.16.12/x64/src/io/io.go:347
golang.org/x/net/http2.readFrameHeader(0xc0001382d8, 0x9, 0x9, 0x18cd520, 0xc00014a120, 0x0, 0x0, 0x0, 0x0)
        /home/runner/work/container-egress-service/container-egress-service/vendor/golang.org/x/net/http2/frame.go:237 +0x89
golang.org/x/net/http2.(*Framer).ReadFrame(0xc0001382a0, 0xc000110960, 0x0, 0x0, 0x0)
        /home/runner/work/container-egress-service/container-egress-service/vendor/golang.org/x/net/http2/frame.go:492 +0xa5
golang.org/x/net/http2.(*clientConnReadLoop).run(0xc0004a9fa8, 0x0, 0x0)
        /home/runner/work/container-egress-service/container-egress-service/vendor/golang.org/x/net/http2/transport.go:1816 +0xd8
golang.org/x/net/http2.(*ClientConn).readLoop(0xc000001680)
        /home/runner/work/container-egress-service/container-egress-service/vendor/golang.org/x/net/http2/transport.go:1738 +0x6f
created by golang.org/x/net/http2.(*Transport).newClientConn
        /home/runner/work/container-egress-service/container-egress-service/vendor/golang.org/x/net/http2/transport.go:694 +0x6c5
```

3. ces deployment yaml:
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: ces-controller-test
  namespace: kube-system
spec:
  replicas: 1
  selector:
    matchLabels:
      app: ces-controller
  template:
    metadata:
      labels:
        app: ces-controller
    spec:
      nodeSelector: 
        kubernetes.io/hostname: node2
      serviceAccountName: ces-controller
      containers:
        - name: ces-controller
          image: f5devcentral/ces-controller:0.5.2
          imagePullPolicy: IfNotPresent
          resources:
            requests:
              cpu: '1'
              memory: 1Gi
            limits:
              cpu: '1'
              memory: 1Gi
          command:
            - /ces-controller
            - --bigip-url=10.0.30.78
            - --bigip-insecure=true
            - --bigip-creds-dir=/ces/bigip-creds
            - --bigip-conf-dir=/ces
            - --v=3
          volumeMounts:
            - name: bigip-creds
              mountPath: "/ces/bigip-creds"
              readOnly: true
            - name: bigip-config
              mountPath: /ces
              readOnly: true
      volumes:
        - name: bigip-creds
          secret:
            secretName: bigip-creds
        - name: bigip-config
          configMap:
            name: ces-controller-configmap
      dnsPolicy: ClusterFirst
      restartPolicy: Always
      terminationGracePeriodSeconds: 30
```

**Comment by myf5** [comment]
Thanks. There is some limit cause current problem.
Will fix the issue in the next release.

**Comment by myf5** [comment]
Removed the initialized controlling parameter and let ces controller to get its ns from the env variable in the deployment.

will release once finishing the internal test.

**Comment by myf5** [comment]
fixed in the release 0.5.3 #7
